### PR TITLE
Avoid GetText deprecation warning when using newer GetText

### PR DESCRIPTION
**Pull request by JasonTrue**
### Summary of changes

(Please note: this summary has been substantially edited to reflect my current understanding per @Sinc63's comments).

Newer versions of gettext are pickier about the Plural-Forms header
structure than before, and will ignore the nplurals behavior unless
there is 1) a rule for which plural form to support and 2) the rule
is in the expected format.

To avoid deprecation warnings, the Plural form rules need to be corrected.
Alternatively, custom implementations of the GetText.Plural behavior can
be used, especially if custom behavior is required. See
[GetText.Plural module documentation](https://hexdocs.pm/gettext/Gettext.Plural.html)

As far as I can tell, the rules used by Timex mirror the rules
implemented by the transitive dependency gettext 0.22 now has on
[Expo](https://github.com/elixir-gettext/expo), so I don't see an
immediate need for a custom implementation of Gettext.Plural.

This PR also removes an incorrectly specified plural rule for
Indonesian.

There may be some remaining issues with Russian and possibly languages
with similar pluralization rules, like Croatian, Polish, and Serbian,
as several unit tests for Russian still fail.

### Checklist

- [x] New functions have typespecs, changed functions were updated
- [x] Same for documentation, including moduledocs
- [x] Tests were added or updated to cover changes
- [x] Commits were squashed into a single coherent commit
- [x] Notes added to CHANGELOG file which describe changes at a high-level

### CODE REVIEW

File: priv/translations/af/LC_MESSAGES/relative_time.po
Comment (JasonTrue) [reply priv/translations/af/LC_MESSAGES/relative_time.po:12]: Ah, that makes more sense. I was perplexed by the messaging about the deprecation warning and assumed that the Plural-Forms header was no longer being supported at all, but on closer reading it's only generating the warning for certain ill-formed headers. In that case, I'll try to replace this PR with a more surgical one that fixes the format rather than removing plural-forms entirely, which I imagine was originally introduced by an GetText editing tool of some sort.

**Comment by JasonTrue** [comment]
Fixes #734.

File: priv/translations/af/LC_MESSAGES/relative_time.po
Comment (maennchen) [reply priv/translations/af/LC_MESSAGES/relative_time.po]: @JasonTrue That is correct:

Either:

* Correct the plural forms header to a complete and valid one
* Or: remove it entirely and use the default known for the language

I recommend the first option since it helps with the tooling as well like Gettext file editors etc.

(I was part of making those changes in `gettext` / `expo`)

**Comment by maennchen** [comment]
> so I don't see an immediate need for a custom implementation of `Gettext.Plural`

That would only be needed if you implement pluralization yourself. Here this is not needed.

File: priv/translations/id/LC_MESSAGES/relative_time.po
Comment (Sinc63) [inline priv/translations/id/LC_MESSAGES/relative_time.po:12]: Correct this to match the one for id in units.po.

**Review by Sinc63 — APPROVED** [review]
This is much simpler and looks great.

File: priv/translations/id/LC_MESSAGES/relative_time.po
Comment (JasonTrue) [reply priv/translations/id/LC_MESSAGES/relative_time.po:12]: Thanks. Because the file wasn't using plurals I had deleted it, but just to be safe I've updated it to match.

**Review by trans — APPROVED** [review]
This looks good. I can't say if the added rules are precisely correct, but it takes care of the warnings properly. Better to merge and fix any plural rules later that arise via usage.

**Review by thiamsantos — APPROVED** [review]

**Comment by travismorrison** [comment]
Hi--any thoughts on when this will get merged? we're mildly blocked on updating deps on one of our apps.

**Comment by KernelMadness** [comment]
Hi, also want to support this PR, looks like it needs only one review from @maennchen

**Review by maennchen — APPROVED** [review]

**Review by bitwalker — APPROVED** [review]

**Comment by bitwalker** [comment]
I'll get this released ASAP

**Comment by warmwaffles** [comment]
@bitwalker any chance for a release with this fix included?

**Comment by bitwalker** [comment]
Done

**Pull request merged**
